lib/pool.dart

```diff
@@ -241,20 +241,20 @@ class Pool {
   ///
   /// This may be called more than once; it returns the same [Future] each time.
   Future close() => _closeMemo.runOnce(() {
-        if (_closeGroup != null) return _closeGroup.future;
+        if (_closeGroup != null) return _closeGroup!.future;
 
         _resetTimer();
 
         _closeGroup = FutureGroup();
         for (var callback in _onReleaseCallbacks) {
-          _closeGroup.add(Future.sync(callback));
+          _closeGroup!.add(Future.sync(callback));
         }
 
         _allocatedResources -= _onReleaseCallbacks.length;
         _onReleaseCallbacks.clear();
 
-        if (_allocatedResources == 0) _closeGroup.close();
-        return _closeGroup.future;
+        if (_allocatedResources == 0) _closeGroup!.close();
+        return _closeGroup!.future;
```

Comment:
@leafpetersen - should this be inferred as non-null here due to the assignment on line 248?

Reply:
Since it's an instance field, there is no guarantee that any of the other calls in the meantime haven't set it back to `null` me thinks

Reply:
What @kevmoo said.  If you want promotion, bind it to a local.